docs/basic/flattening.md

```diff
@@ -0,0 +1,134 @@
+- Scar removal (remove long, thin, bright streaks in the data)
+- Zero the average height (lower the image by its median height) to make the background roughly centred at zero nm
+- Masking (detect objects on the surface and flatten the image again, ignoring the data on the surface)
+- Secondary flattening (re-process the data using the mask to tell us where the background is)
+- Gaussian filter (to remove jitteryness / noise)
+
+![flattening pipeline](../_static/images/flattening/flattening_pipeline.png)
+
+## Row alignment
+
+The first step in the flattening process is **row alignment**. Row alignment is a process that adjusts the height of
+each row of the image so that they all share the same median height value. This gets rid of some of the horizontal
+banding. This leaves an image where the rows are aligned, but the image still has a clear tilt.
+
+![row alignment](../_static/images/flattening/flattening_align_rows.png)
+
+## Tilt removal
+
+After row alignment, tilt removal is applied. This is a simple process of fitting a plane to the image and subtracting
+it, resulting in a mostly flat image, however as you can see in the following image, it's not perfect.
+Two images are provided here, one with the full z-range and one with an adjusted z-range to show the remaining
+artefacts better.
+
+![tilt_removal_full_zrange](../_static/images/flattening/flattening_tilt_removal_full_zrange.png)
+
+![tilt removal_better_viewing](../_static/images/flattening/flattening_tilt_removal.png)
+
+## Polynomial removal
+
+After the tilt, we remove the polynomial trends. In some images, there is also quadratic or occasionally cubic bowing to
+the image too. We remove this by fitting a two dimensional quadratic polynomial to the image (in the horizontal
+direction), and subtracting it from the image. We then do the same for a nonlinear polynomial (z = a*x*y) to eliminate
+“saddle” trends in the data. We could do all of these at the same time, but we like to be able to see the iterative
+differences.
+
+## Scar removal
+
+We then run scar removal on the image. This is a special function that detects scars (long, thin, bright streaks in the
+data, caused by physical problems in the AFM process). We are using a different image here since our lovely
+minicircles.spm image doesn’t have any scars.
+
+![scarred image](../_static/images/flattening/flattening_scarred_image.png)
+
+![scar removed](../_static/images/flattening/flattening_scar_removed.png)
+
+**Note that scar removal can distort data, and it’s best to take data without scars if you can.**
+
+## Zero the average height
+
+![height zeroing](../_static/images/flattening/flattening_height_zeroing.png)
+
+We then lower the image by its median height which causes the background of the image to be roughly centred at zero nm.
+This is important since both the AFM and these processing steps can cause your background height to not be zero, which
+would mean you would get the wrong heights for your data.
```

Comment:
```suggestion
We then lower the image by its mean height which causes the background of the image to be roughly centred at zero nm.
If this function is provided a foreground mask such as in the second iteration of flattening, this function zeros the data only on the background data.
Data zeroing is important since the raw AFM heights are relative, and these processing steps can shift the background height away from zero, so this makes it easier to obtain comparative height metrics.
```

Btw the averaging code is all based on means not medians:
```
if mask is None:
    mask = np.zeros_like(image)
mean = np.mean(image[mask == 0])
LOGGER.debug(f"[{self.filename}] : Zero averaging background : {mean} nm")
return image - mean
```

Reply:
❤️